src/Util/Message.cc

```diff
@@ -740,6 +698,85 @@ CARTA::RemoteFileRequest Message::RemoteFileRequest(int32_t file_id, const strin
+    return response;
+}
+
+CARTA::FileInfoExtended Message::AddComputedEntry(
+    CARTA::FileInfoExtended& response, std::string name, const std::string& value, CARTA::EntryType type, double numeric_value) {
+    auto entry = response.add_computed_entries();
+    entry->set_name(name);
+    entry->set_value(value);
+    entry->set_entry_type(type);
+    entry->set_numeric_value(numeric_value);
+    return response;
+}
+
+CARTA::ImportRegionAck Message::AddImportedRegion(CARTA::ImportRegionAck& import_ack, int region_id, CARTA::RegionType region_type,
+    std::vector<CARTA::Point> control_points, float region_rotation, CARTA::RegionStyle region_style) {
+    // Set CARTA::RegionInfo
+    CARTA::RegionInfo region_info;
+    region_info.set_region_type(region_type);
+    *region_info.mutable_control_points() = {control_points.begin(), control_points.end()};
+    region_info.set_rotation(region_rotation);
+
+    // Add info and style to import_ack; increment region id for next region
+    (*import_ack.mutable_regions())[region_id] = region_info;
+    (*import_ack.mutable_region_styles())[region_id] = region_style;
+
+    return import_ack;
+}
+
+CARTA::SpatialProfileData Message::AddProfile(CARTA::SpatialProfileData& message, std::string coordinate, int start, int end,
+    casacore::Float* profile_data, size_t profile_size, int mip) {
+    // add SpatialProfile to message
+    auto spatial_profile = message.add_profiles();
+    spatial_profile->set_coordinate(coordinate);
+    // Should these be set to the rounded endpoints if the data is downsampled or decimated?
+    spatial_profile->set_start(start);
+    spatial_profile->set_end(end);
+    spatial_profile->set_raw_values_fp32(profile_data, profile_size);
+    spatial_profile->set_mip(mip);
+    return message;
+}
+
+CARTA::SpatialProfileData Message::AddProfile(CARTA::SpatialProfileData& message, int32_t file_id, int32_t region_id, int32_t start,
+    std::vector<float>& profile, std::string& coordinate, int32_t mip, CARTA::ProfileAxisType axis_type, casacore::Quantity& increment) {
+    auto profile_size = profile.size();
+    int end = profile_size - 1;
+    float crpix = profile_size / 2;
+    float cdelt = increment.getValue();
+    float crval = (axis_type == CARTA::ProfileAxisType::Offset ? 0.0 : crpix * cdelt);
+    std::string unit = increment.getUnit();
+    message.set_file_id(file_id);
+    message.set_region_id(region_id);
+    auto* spatial_profile = message.add_profiles();
+    spatial_profile->set_start(start);
+    spatial_profile->set_end(end);
+    spatial_profile->set_raw_values_fp32(profile.data(), profile.size() * sizeof(float));
+    spatial_profile->set_coordinate(coordinate);
+    spatial_profile->set_mip(mip);
+    auto* profile_axis = spatial_profile->mutable_line_axis();
+    profile_axis->set_axis_type(axis_type);
+    profile_axis->set_crpix(crpix);
+    profile_axis->set_crval(crval);
+    profile_axis->set_cdelt(cdelt);
+    profile_axis->set_unit(unit);
```

Comment:
There should be a separate wrapper for adding a profile axis (sub-)submessage to the spatial profile submessage. Then this alternative `AddProfile` can be eliminated. See general remarks re: return values -- all these wrappers that add submessages should return the *submessage*, not the parent message. Then you can chain operations (e.g. adding a submessage to a submessage, or setting another field on it).